transcriptic/commands.py

```diff
@@ -1476,6 +1476,72 @@ def run_protocol(api, manifest, protocol, inputs, view=False, dye_test=False):
             return
 
 
+def execute(
+    autoprotocol,
+    api,
+    workcellid,
+    deviceset,
+    timelimit,
+    partitiongroupsize,
+    partitionhorizon,
+    partitioningswapdeviceid,
+):
+    # Get the autoprotocol
+    autoprotocol_str = autoprotocol.read()
+    try:
+        autoprotocol_dict = json.loads(autoprotocol_str)
+    except json.decoder.JSONDecodeError as err:
+        print("Error decoding autoprotocol json: ")
+        print(err)
+        return
+
+    # Define the initial payload
+    payload = {"autoprotocol": autoprotocol_dict, "timeLimit": f"{timelimit}:second"}
+
+    if deviceset:
+        device_str = deviceset.read()
+        try:
+            device_json = json.loads(device_str)
+            payload["deviceSet"] = device_json
+        except json.decoder.JSONDecodeError as err:
+            print("Error decoding device set json: ")
+            print(err)
+            return
+    elif workcellid:
+        payload["workcellIdForDeviceSet"] = f"{workcellid}-mcx1"
+    else:
+        payload["workcellIdForDeviceSet"] = "wc0-mcx1"
+
+    if partitiongroupsize is not None:
+        payload["partitionGroupSize"] = partitiongroupsize
+
+    if partitionhorizon is not None:
+        payload["partitionHorizon"] = f"{partitionhorizon}:second"
+
+    if partitioningswapdeviceid is not None:
+        payload["partitioningSwapDeviceId"] = partitioningswapdeviceid
+
+    # Clean api end point
+    if api[-1] == "/":
+        clean_api = api[0:-1]  # remove trailing slash
+    else:
+        clean_api = api
+
+    # POST to workcell
+    test_run_endpoint = f"{clean_api}/testRun"
+    print("Sending request...")
+    res = requests.post(test_run_endpoint, json=payload)
+    try:
+        res_json = json.loads(res.content)
+        if res_json["success"]:
+            print("Success. View dashboard for scheduling job.")
```

Comment:
Maybe putting the url with the correct port can be nice. as we have to add the :9000
On another note, can we redirect to 9000 automatically in the lab repo that would be great :D